test/snippet/core/algorithm/configuration.cpp

```diff
@@ -1,88 +1,96 @@
+#include <iostream>
+
+//! [configuration_setup]
 #include <seqan3/core/algorithm/configuration.hpp>
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
 
 using namespace seqan3;
 
-struct bar
+enum struct my_id : int
 {
-    float value = 0;
+    bar_id,
+    foo_id
 };
 
-struct foo
+struct bar : public pipeable_config_element
 {
-    int value = 0;
+    static constexpr my_id id{my_id::bar_id};
+
+    bar()                        = default;
+    bar(bar const &)             = default;
+    bar(bar &&)                  = default;
+    bar & operator=(bar const &) = default;
+    bar & operator=(bar &&)      = default;
+    ~bar()                       = default;
+
+    bar(float v) : value(v)
+    {}
+
+    float value;
 };
 
-struct with_foo_adaptor : public detail::configuration_fn_base<with_foo_adaptor>
+template <typename t>
+struct foo : public pipeable_config_element
```

Comment:
```suggestion
struct foo : public pipeable_config_element  // define a templatized configuration element 'foo'
```

Reply:
I don't know. I mean the snippet is inserted in a documentation test and there is a proper documentation for it.

Reply:
Hm, yes but I think usually people like to look at examples and skip the text (unless they really want to dive into the topic), so I always prefer snippets to be mostly understandable without the wrapped text. But this is your call then.